### DISCUSSION TITLE
Utility to parse `string` to `0x${string}`

### DISCUSSION THREAD

**Discussion by BrennerSpear** [wevm/wagmi#1725] [Q&A]
Some of the params require type `0x${string}`, but sometimes I'm getting an address from somewhere other than from wagmi, so the address is just a `string`. What's the safest way to parse + typecast to `0x${string}`? Does wagmi have a utility function for this?

**Comment by peterferguson** [comment]
Think the best way is either `zod` or using `ethers` `getAddress` as it returns `0x${string}`.

I usually find myself doing 

```ts
import { getAddress } from '@ethersproject/address'

...

const address = !!addressFromElsewhere ?  getAddress(addressFromElsewhere) : undefined
// ^? `0x${string}` | undefined
```

**Comment by Raiden1411** [comment]
You can use the `as` keyword from typescript to declare it as `0x${string}`
You could also change it use string instead of `0x${string}`
Check the abitype config section on the readme file.
